test/extended/authentication/component_proxy_helpers.go

```
@@ -0,0 +1,826 @@
+			if !ok {
+				return false, nil
+			}
+			return d.Status.ReadyReplicas > 0, nil
+		},
+	)
+	if err != nil {
+		return "", "", nil, "", cleanups, fmt.Errorf("squid proxy deployment did not become ready: %w", err)
+	}
+
+	serviceHost := fmt.Sprintf("%s.%s.svc.cluster.local", squidServiceName, namespace)
+	httpProxyURL = fmt.Sprintf("http://%s:%d", serviceHost, squidHTTPPort)
```

golangci-lint (2.12.2)

[error] host:port in url should be constructed with net.JoinHostPort and not directly with fmt.Sprintf

(nosprintfhostport)